src/api/router/RouterHistory.ts

```diff
@@ -0,0 +1,28 @@
+import { ListLikeObvserver } from "../../internals/ListLikeObserver";
+
+export type HistoryItem = { route: string, payload?: any };
+
+export class RouterHistory<T extends HistoryItem = HistoryItem> extends ListLikeObvserver<T> {
```

Comment:
An observer must never modify the obversvee. These methods should be router internals, or RouterHistory class it wraps, not extend the observer.